src/libical/icalcomponent.c

```diff
@@ -2461,8 +2461,12 @@ void icalcomponent_set_due(icalcomponent *comp, struct icaltimetype v)
         icalproperty_set_duration(dur_prop, dur);
     }
 
-    if (due_prop && (tzid = icaltime_get_tzid(v)) != NULL && !icaltime_is_utc(v)) {
-        icalproperty_set_parameter(due_prop, icalparameter_new_tzid(tzid));
+    if (due_prop) {
+        if ((tzid = icaltime_get_tzid(v)) != NULL && !icaltime_is_utc(v)) {
+            icalproperty_set_parameter(due_prop, icalparameter_new_tzid(tzid));
+        } else {
+            icalproperty_remove_parameter_by_kind(due_prop, ICAL_TZID_PARAMETER);
```

Comment:
Not that its critical, but couldn't the remove_param line simply be added after line #2453?  This would more closely match what set_dtstart looks like

Reply:
You're right. I initially didn't do that because it seemed like calling `icalproperty_remove_parameter_by_kind` every time was unnecessary, but it's better with your suggestion.